server/server.js

```diff
@@ -172,6 +184,40 @@ app.post('/genotet/upload', upload.single('file'), function(req, res) {
   });
 });
 
+/**
+ * POST request is not used as it conflicts with jsonp.
+ */
+app.post('/genotet/user', function(req, res) {
+  console.log('POST user');
+
+  var type = req.body.type;
+  var userInfo = {};
+  var response;
+  switch (type) {
+    case user.QueryType.SIGHUP:
+      userInfo = {
+        email: req.body.email,
+        username: req.body.username,
+        password: req.body.password,
+        confirmed: req.body.confirmed
+      };
+      response = user.signUp(userPath, userInfo);
+      break;
+    case user.QueryType.SIGNIN:
+      userInfo = {
+        username: req.body.username,
+        password: req.body.password
+      };
+      response = user.signIn(userPath, userInfo);
+      break;
+  }
+  res.header('Access-Control-Allow-Origin', '*');
+  res.json({
+    success: true,
```

Comment:
On error we status 500 and send back error object. Take a look at app.get's subroutines.